File: advent2021/day6/main.py
import os, sys

def part1():
    with open(os.path.join(sys.path[0], "input.txt"), "r") as f:
        days = [int(c) for c in [x for x in f][0].split(",")]
        for _ in range(18):
            temp = []
            for x in days:
                if x == 0:
                    temp.append(8)
                    temp.append(6)

                else:
                    temp.append(x - 1)

            days = temp
    return len(days)
print(part1())

def part2():
    with open(os.path.join(sys.path[0], "input.txt"), "r") as f:
        days = [int(c) for c in [x for x in f][0].split(",")]
        fish = {}
        for i in range(9):
            fish[i] = days.count(i)
        for _ in range(256):
            new_fish = fish[0]
            for c in range(0, 8):
                fish[c] = fish[c + 1]
            fish[8] = new_fish
            fish[6] += new_fish

    return sum(fish.values())

print(part2())

File: advent2021/day6/main.rs
use std::fs;
use std::collections::HashMap;

fn main() {
    let contents = fs::read_to_string("input.txt").expect("Unable to read the file");
    let mut days = Vec::<i64>::new();
    for c in contents.split(",") {
        days.push(c.trim().parse().expect(""));
    }
    let mut fish = HashMap::new();
    for i in 0..9 {
         let _ = *fish.entry(i).or_insert(0);
    }
    for day in &days {
        *fish.entry(*day).or_insert(0) += 1;
    }
    
    for _ in 1..=256{
        let temp = fish[&0];
        for c in 0..8 {
            let count = fish[&(&c + 1)];
            fish.insert(c, count);
        }
        *fish.get_mut(&8).unwrap() = temp;
        *fish.get_mut(&6).unwrap() += temp;
    }
    println!("{}", fish.values().sum::<i64>());


}
